### PR TITLE
Display tags column

### PR DESCRIPTION
**Pull request by DarkDrek**
Implements #38 

This just reserves enough space to display all tag values which may not work when there are a lot of tags.
In a previous commit a had a simple logic to just display ... after the end of the space was reached but I'm not sure if that is the right approach?

### CODE REVIEW

**Comment by DarkDrek** [comment]
I've rebased it.

File: src/cli.rs [hunk 1]
```diff
@@ -399,21 +403,25 @@ impl DisplayOptions {
...
         age_width,
         spent_width,
         status_width,
         description_width,
         project_width,
+        tags_width,
         has_spent_time,
         has_priorities,
         has_projects,
+        has_tags,
```
Comment (phaazon) [inline src/cli.rs:424]: This is just a note for myself: I really think we should get rid of this huge fold accumulator and use a data type. It’s becoming hard to follow what is what. :D

File: src/cli.rs [hunk 2]
```diff
@@ -525,18 +539,29 @@ impl DisplayOptions {
     task.project().map(UnicodeWidthStr::width)
   }
 
+  /// Guess the width required to represent the task tags.
+  fn guess_tags_width(task: &Task) -> usize {
+    task
+      .tags()
+      .intersperse(", ")
+      .map(UnicodeWidthStr::width)
+      .sum()
```
Comment (phaazon) [inline src/cli.rs:548]: I’m not a huge fan of this — we’re going to allocate several time / recompute several times the same string. I think we should move away from doing things this way.

File: src/cli.rs [hunk 3]
```diff
@@ -676,6 +721,15 @@ fn display_task_inline(config: &Config, uid: UID, task: &Task, opts: &DisplayOpt
   display_description(config, opts, status, task_name);
 }
 
+/// Display the tags by respecting the allowed tags column size.
+fn display_tags(task: &Task, opts: &DisplayOptions) {
+  print!(
+    " {tags:<tags_width$}",
+    tags = task.tags().intersperse(", ").collect::<String>(),
```
Comment (phaazon) [inline src/cli.rs:728]: Same remark as above.

**Review by phaazon — CHANGES_REQUESTED** [review]

File: src/cli.rs (hunk 1 above)
Comment (LordMZTE) [reply src/cli.rs:424]: yeah ive had some trouble understanding that before

File: src/cli.rs (hunk 2 above)
Comment (LordMZTE) [reply src/cli.rs:548]: this could be solved by summing up the width of all tags, and then adding the width of the separator `n-1` times

**Comment by phaazon** [comment]
As with #59, I’ll take it over. Thanks!
